File: bench/src/collatz.py
import sys

def collatz(n):
    i = 0
    while n > 1:
        if n % 2 == 0:
            n = n / 2
        else:
            n = n * 3 + 1
        i += 1
    return i

def max_len(n):
    m = 0
    for i in range(n):
        m = max(m, collatz(i))
    return m;

if __name__ == '__main__':
    print(max_len(1000000))

File: bench/src/collatz.java
class Collatz {

	private static int collatz(long n) {
		int i = 0;
		while (n > 1) {
			i++;
			if (n % 2 == 0) {
				n = n / 2;
			} else {
				n = 3 * n + 1;
			}
		}
		return i;
	}

	private static int maxlen(long n) {
		int max = 0;
		for (int i = 1; i < n; i++) {
			int m = collatz(i);
			if (m > max) {
				max = m;
			}
		}
		return max;
	}

	public static void main(String[] args) {
		System.out.println(maxlen(1000000));
	}

}
